docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,192 @@
+.. _chat_dataset_usage_label:
+
+=============
+Chat Datasets
+=============
+
+Chat datasets involve multi-turn conversations (multiple back-and-forths) between user and assistant.
+This is more structured than freeform text association that models are typically pre-trained with,
+where they learn to simply predict the next token instead of responding accurately to the user.
+
+The primary entry point for fine-tuning with chat datasets in torchtune is the :func:`~torchtune.datasets.chat_dataset`
+builder. This lets you specify a local or Hugging Face dataset that follows the chat data format
```

Comment:
For example, to use the X chat dataset from HuggingFace, use the following config...

^^ ideally one which needs just enough customisation to show the user how the parameters work - I think this would give sufficient info to a fair amount of our users already without needing to deepdive. I think Evan's comments on the other PR are good - lead with examples and detail for people who perhaps don't grok just from examples